client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -40,26 +42,30 @@ function filterPatternsByType( patterns: Pattern[], type: PatternTypeFilter ) {
 	} );
 }
 
-type PatternsCategoryPageProps = {
+type PatternLibraryProps = {
 	category: string;
+	categoryGallery: CategoryGalleryFC;
 	isGridView?: boolean;
 	patternGallery: PatternGalleryFC;
 	patternTypeFilter: PatternTypeFilter;
+	searchTerm?: string;
 };
 
-export const PatternsCategoryPage = ( {
+export const PatternLibrary = ( {
 	category,
+	categoryGallery: CategoryGallery,
 	isGridView,
 	patternGallery: PatternGallery,
 	patternTypeFilter,
-}: PatternsCategoryPageProps ) => {
+	searchTerm: urlQuerySearchTerm,
+}: PatternLibraryProps ) => {
 	const locale = useLocale();
 	// Helps prevent resetting the search input if a search term was provided through the URL
 	const isInitialRender = useRef( true );
 	// Helps reset the search input when navigating between categories
 	const [ searchFormKey, setSearchFormKey ] = useState( category );
```

Comment:
Why do we need it? We don't use `key` inside `PatternsHeader`

Reply:
The `initialSearchTerm` prop we pass to `PatternsHeader` only takes effect on the first render. When users navigate between categories, we want React to think it's a "first render" again because we need to reset the search input. Passing a new `key` prop has that effect.